packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+        converterKeysTableName,
+        axiosTimeoutSeconds,
+        log,
+      });
+
+      const formattedXML = formatXML(res.data);
+      const fileName = createFileName({ cxId, patientId, resources, dateFrom, dateTo });
+
+      await s3Client
+        .putObject({
+          Bucket: bucketName,
+          Key: fileName,
+          Body: formattedXML,
+          ContentType: "application/xml",
+        })
+        .promise();
+
+      const convertUrl = await convertDoc({ fileName, conversionType });
+
+      return convertUrl;
+    } catch (err) {
+      console.log(
+        `Error processing bundle for patient: ${patientId} with resources ${resources}; ${err}`
+      );
+      capture.error(err, {
+        extra: {
+          patientId,
+          resources,
+          dateFrom,
+          dateTo,
+          context: lambdaName,
+        },
+      });
+      throw err;
+    }
+  }
+);
+
+const formatXML = (xml: string): string => {
+  const parser = new DOMParser();
+  const document = parser.parseFromString(xml, "text/xml");
+
+  const structuredBody = document.getElementsByTagName("structuredBody")[0];
+  const components = Array.from(structuredBody.getElementsByTagName("component"));
+
+  const problemListComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Problem list";
+  });
+
+  const treatmentPlanComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Treatment Plan";
+  });
+
+  const medicationComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "History of Medication use";
+  });
+
+  if (problemListComponent) {
+    const problemListTable = problemListComponent?.getElementsByTagName("table")[0];
+
+    addIcd10Codes(problemListComponent, problemListTable, document);
+  }
+
+  if (treatmentPlanComponent) {
+    const firstComponent = components[0];
+    structuredBody.insertBefore(treatmentPlanComponent, firstComponent);
+  }
+
+  if (problemListComponent) {
+    const reshapedComponents = Array.from(structuredBody.getElementsByTagName("component"));
+    const secondComponent = reshapedComponents[1];
+    structuredBody.insertBefore(problemListComponent, secondComponent);
+  }
+
+  if (medicationComponent) {
+    const reshapedComponents = Array.from(structuredBody.getElementsByTagName("component"));
+    const thirdComponent = reshapedComponents[2];
+    structuredBody.insertBefore(medicationComponent, thirdComponent);
+  }
+
+  return document.toString();
+};
+
+const addIcd10Codes = (
+  problemListComponent: Element,
+  problemListTable: Element,
+  document: Document
+) => {
+  addCodesToColumn(problemListTable, document);
+
+  const tableRows = problemListTable.getElementsByTagName("tbody")[0].getElementsByTagName("tr");
```

Comment:
we should check for the access of index 0, it could be `undefined`, no?

also, `tableRows could be `undefined` but `addCodesToRows` requires it to be not undefined

Reply:
The linter was getting these errors before so now i resolved it